deployment/simProposalExecution.js

```diff
@@ -0,0 +1,116 @@
+require('@nomiclabs/hardhat-ethers');
+const Helper = require('../test/testHelpers/hardhat');
+
+let kyberGov;
+
+let forkParams = {};
+let admin;
+let daoOperator;
+let kyberGovAddress;
+let executorAddress;
+let voterAddresses = [
+  '0xf977814e90da44bfa03b6295a0616a897441acec',
+  '0x3Be35d6C5FFeAe62CB3f6CB8a23653b6501A060d',
+  '0x0bfEc35a1A3550Deed3F6fC76Dde7FC412729a91',
+  '0x06890D4c65A4cB75be73D7CCb4a8ee7962819E81',
+];
+const chainIdToAddresses = {
+  1: {
+    kyberGov: '0x7ec8fcc26be7e9e85b57e73083e5fe0550d8a7fe',
+    daoOperator: '0xe6a7338cba0a1070adfb22c07115299605454713',
+  },
+  3: {
+    kyberGov: '0xef5a1404E312078cd16B7139a2257eD3bb42F787',
+    daoOperator: '0xDdF05698718bA8ed1c9abA198d38a825A64D69e2',
+  },
+  31337: {
+    kyberGov: '0x7ec8fcc26be7e9e85b57e73083e5fe0550d8a7fe',
+    daoOperator: '0xe6a7338cba0a1070adfb22c07115299605454713',
+  }
+};
+
+let oneEth;
+
+let startTimestamp;
+let endTimestamp;
+let proposalId;
+
+task('simProposalExecution', 'simulate execution of existing proposal')
+  .addParam('id', 'Proposal ID')
+  .setAction(async (taskArgs) => {
+    getForkParams();
+    proposalId = taskArgs.id;
+
+    // attempt mainnet forking
+    try {
+      await network.provider.request({
+        method: 'hardhat_reset',
+        params: [forkParams],
+      });
+
+      oneEth = ethers.constants.WeiPerEther;
+      [admin] = await ethers.getSigners();
+      getAddresses(await admin.getChainId());
+      kyberGov = await ethers.getContractAt('KyberGovernance', kyberGovAddress);
+    } catch (e) {
+      console.log(e);
+      process.exit(1);
+    }
+
+    daoOperator = await Helper.impersonateAcc(network, ethers.provider, daoOperator, admin);
+    console.log('Fetching proposal details...');
+    // get proposal timestamps
+    await getProposalDetails();
+
+    // fast forward time
+    await Helper.mineNewBlockAt(network, startTimestamp + 1);
```

Comment:
if proposal is started, this would be failed.
```console
$ yarn hardhat simProposalExecution --id 1
yarn run v1.22.4
$ /Users/nguyenducquang/workspace/smart-contract/kyber/dao_sc/node_modules/.bin/hardhat simProposalExecution --id 1
Fetching proposal details...
An unexpected error occurred:

InvalidInputError: Timestamp 1623136028 is lower than previous block's timestamp 1623308779
    at EvmModule._mineAction (/Users/nguyenducquang/workspace/smart-contract/kyber/dao_sc/node_modules/hardhat/src/internal/hardhat-network/provider/modules/evm.ts:94:15)
    at process._tickCallback (internal/process/next_tick.js:68:7)
error Command failed with exit code 1.

```

Reply:
there are a number of proposal states to handle... i'm working with the assumption is that we only will be running this script after the multisig has submitted the proposal to verify the proposal data.

Reply:
u can pin an older block number where the proposal isn't active yet

Reply:
nvm, updated script to handle the different proposal states